legacy/build-deploy.sh

```diff
@@ -11,10 +11,10 @@ REGISTRY=$REGISTRY
 NAMESPACE=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
 LAGOON_VERSION=$(cat /lagoon/version)
 
-if [ ! -z "$LAGOON_PROJECT_VARIABLES" ]; then
-  INTERNAL_REGISTRY_URL=$(jq --argjson data "$LAGOON_PROJECT_VARIABLES" -n -r '$data | .[] | select(.scope == "internal_container_registry") | select(.name == "INTERNAL_REGISTRY_URL") | .value' | sed -e 's#^http://##' | sed -e 's#^https://##')
-  INTERNAL_REGISTRY_USERNAME=$(jq --argjson data "$LAGOON_PROJECT_VARIABLES" -n -r '$data | .[] | select(.scope == "internal_container_registry") | select(.name == "INTERNAL_REGISTRY_USERNAME") | .value')
-  INTERNAL_REGISTRY_PASSWORD=$(jq --argjson data "$LAGOON_PROJECT_VARIABLES" -n -r '$data | .[] | select(.scope == "internal_container_registry") | select(.name == "INTERNAL_REGISTRY_PASSWORD") | .value')
+if [ ! -z "$LAGOON_ENVIRONMENT_VARIABLES" ]; then
```

Comment:
These are set in the `remote-controller` and are added to the project variables.

The API used to set these, but since we stopped using harbor in core, they are no longer provided by the API.

Since we can't rely on the controller being updated before the build image being updated, we'd need to adjust this to check both project and environment variables in a similar way to how you used the `apiEnvVarCheck` function, and then we can work on changing the [remote-controller code to use the environment variables](https://github.com/uselagoon/remote-controller/blob/347cd933e06ac5f91ed5e12776dc792b7f8a0590/internal/controllers/v1beta2/build_helpers.go#L620C25-L628) instead of project.

Reply:
So we can also fully deprecate that enum in the API?

Reply:
we should probably nuke any columns that contain it in the api-db too